pkg/client/core/store/filesystem/application_filesystem.go

```diff
@@ -14,14 +16,47 @@ type applicationStore struct {
 	fs    *afero.Afero
 }
 
+// Helper for optional resources
+func addOperationIfNotEmpty(operations store.Operations, root, filePath string, content []byte) {
+	if len(content) == 0 {
+		return
+	}
+
+	operations.StoreBytes(path.Join(root, filePath), content)
+}
+
 // SaveApplication applies the application to the file system
 func (s *applicationStore) SaveApplication(application *client.ScaffoldedApplication) (*store.Report, error) {
-	report, err := store.NewFileSystem(path.Join(s.paths.BaseDir, application.ApplicationName), s.fs).
-		StoreBytes(fmt.Sprintf("%s.yaml", application.ApplicationName), application.KubernetesResources).
-		StoreBytes(fmt.Sprintf("%s-application.yaml", application.ApplicationName), application.ArgoCDResource).
-		Do()
+	absoluteApplicationDir := path.Join(s.paths.BaseDir, application.ApplicationName)
+	relativeApplicationBaseDir := config.DefaultApplicationBaseDir
+	relativeApplicationOverlayDir := path.Join(config.DefaultApplicationOverlayDir, application.Environment)
+
+	err := s.fs.MkdirAll(path.Join(absoluteApplicationDir, relativeApplicationBaseDir), 0o744)
```

Comment:
By default `store` will create directories that do not exist. I don't think you should have to do this?

Reply:
Are you sure? It created the base directory, but not the subdirectories. I can try again, but the whole reason we put that line there was because we were missing a folder